src/main/java/org/dcsa/ebl/service/impl/TransportDocumentTOServiceImpl.java

```diff
@@ -226,18 +223,37 @@
         }
     }
 
-    private Mono<Void> updateTransportDocumentWithTransportPlan(TransportDocumentTO transportDocumentTO) {
-//                                                                    return Flux.concat(
-//                                                                            ,shipmentService.findByCarrierBookingReference(carrierBookingReference)
-//                                                                                    .next()
-//                                                                                    .flatMap(shipment -> {
-//                                                                                        shipmentTransportService.findByShipmentIDOrderBySequenceNumber(shipment.getId())
-//                                                                                                .next()
-//                                                                                                .flatMap(shipmentTransport -> xxx)
-//                                                                                    })
-//                                                                    )
-//                                                                    .count()
-//                                                                    .then();
+    private Mono<Void> updateTransportDocumentWithTransportPlan(String carrierBookingReference, TransportDocumentTO transportDocumentTO) {
+        TransportPlan transportPlan = new TransportPlan();
+        transportDocumentTO.setTransportPlan(transportPlan);
+        return Flux.concat(
+                // Update ShipmentLocations on TransportPlan
+                shipmentLocationService.findAllByCarrierBookingReference(carrierBookingReference)
+                        .concatMap(shipmentLocation -> {
+                            ShipmentLocationTO shipmentLocationTO = MappingUtil.instanceFrom(shipmentLocation, ShipmentLocationTO::new, AbstractShipmentLocation.class);
+                            return locationService.findById(shipmentLocationTO.getLocationID())
+                                    .flatMap(location -> {
+                                        if (location.getAddressID() != null) {
+                                            return addressService.findById(location.getAddressID())
+                                                    .map(location::toLocationTO);
+                                        }
+                                        return Mono.just(location.toLocationTO(null));
+                                    })
```

Comment:
Maybe we should move this logic into the `LocationService`?  It is not the first time we need it.

E.g. a `findTOById`